controllers/drplacementcontrol_controller.go

```diff
@@ -750,7 +750,18 @@ func (d *DRPCInstance) createPVManifestWorkForRestore(newPrimary string) error {
 	pvMWName := d.mwu.BuildManifestWorkName(rmnutil.MWTypePV)
 
 	existAndApplied, err := d.mwu.ManifestExistAndApplied(pvMWName, newPrimary)
-	if err != nil && !errors.IsNotFound(err) {
+	if err != nil {
+		if errors.IsNotFound(err) {
+			d.log.Info("Restore PVs", "newPrimary", newPrimary)
+
+			if err = d.restore(newPrimary); err != nil {
+				return err
+			}
+
+			// Just restored, wait for MW to generate change event and move to applied
+			return fmt.Errorf("waiting for PV manifestwork (%s) status to be applied", pvMWName)
```

Comment:
(nit) maybe changed the text to: 
`Created PV restore MW. waiting for PV manifestwork (%s) status to be applied`
just to distinguished it from line 774

Reply:
Modified the error message, so that it is different than the one above.

Reply:
Should it be the other way around.  IOW, line 762 is actually when we start the wait because we just created MW for restore?

Reply:
Ack! fixing it. Was dealing with go 1.16 on F33 a well and goofed this up.

Reply:
@BenamarMk Fixed this, but ran into another test failure (due to this PR), where drpc update errors out as object is updated. Fixed the test as well, so there are more changes in the test file that needs a relook (other than this message fix).